File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOFlashBinInfoModel.swift
//
//  IDOFlashBinInfoModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Get Font Library Information event number
@objcMembers
public class IDOFlashBinInfoModel: NSObject, IDOBaseModel {
    /// Status: 0 - Normal, 1 - Invalid font, checksum error, 2 - Version mismatch
    public var status: Int
    /// Font library version
    private let version: Int
    /// Matching version required by the firmware
    public var matchVersion: Int
    /// Font library checksum code
    public var checkCode: Int

    enum CodingKeys: String, CodingKey {
        case status = "status"
        case version = "version"
        case matchVersion = "match_version"
        case checkCode = "check_code"
    }

    public init(status: Int, matchVersion: Int, checkCode: Int) {
        self.status = status
        self.version = 0
        self.matchVersion = matchVersion
        self.checkCode = checkCode
    }

    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOFlashBinInfoModel.kt
//
//  IDOFlashBinInfoModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Get Font Library Information event number

open class IDOFlashBinInfoModel(status: Int, matchVersion: Int, checkCode: Int) : IDOBaseModel {

    /// Status: 0 - Normal, 1 - Invalid font, checksum error, 2 - Version mismatch
    @SerializedName("status")
    var status: Int = status

    /// Font library version
    @SerializedName("version")
    private var version: Int = 0

    /// Matching version required by the firmware
    @SerializedName("match_version")
    var matchVersion: Int = matchVersion

    /// Font library checksum code
    @SerializedName("check_code")
    var checkCode: Int = checkCode


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    